integration/testrunner/db.go

```diff
@@ -82,34 +99,40 @@ func makeBigQueryClient(ctx context.Context, tb testing.TB, projectID string) *b
 	return client
 }
 
-// This calls the database to check that an audit log exists. It uses the retries that are specified in the Config
-// file. This method assumes that only a single audit log will match, which constitutes success.
-func queryIfAuditLogExistsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string) {
-	tb.Helper()
-
-	queryIfAuditLogsExistWithRetries(ctx, tb, bqQuery, cfg, testName, 1)
-}
-
 // This calls the database to check that an audit log exists. It uses the retries that are specified in the Config
 // file. This method allows for specifying how many logs we expect to match, in order to handle streaming use cases.
-func queryIfAuditLogsExistWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string, expectedNum int64) {
+func queryAuditLogsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string, wantNum int) []*loggingpb.LogEntry {
 	tb.Helper()
-
-	b := retry.NewExponential(cfg.LogRoutingWait)
+	var logEntries []*loggingpb.LogEntry
+	b := retry.NewConstant(cfg.LogRoutingWait)
 	if err := retry.Do(ctx, retry.WithMaxRetries(cfg.MaxDBQueryTries, b), func(ctx context.Context) error {
-		found, err := queryIfAuditLogExists(ctx, tb, bqQuery, expectedNum)
-		if found {
-			// Early exit retry if queried log already found.
+		results, err := queryAuditLogs(ctx, tb, bqQuery)
+		if err != nil {
+			return retry.RetryableError(fmt.Errorf("query error: %w", err))
+		}
+		// Early exit retry if queried log already found.
+		if len(results) >= wantNum {
+			logEntries = results
 			return nil
```

Comment:
Here if you find `len(results) > wantNum`, you can return non-retryable error instead. That way, you can save `checkNum` everywhere.

Reply:
Is it unclear that we do such check in this helper function?

Reply:
Not unclear to me. The func accepts a `wantNum` - meaning we expect that number of logs. If it's more than that, it's a test error. If we still don't get that number of logs after retry timeout, it's also a test error. What's unclear about it?

Reply:
Probably because the function name is `queryAuditLogs`, can I move all the test validation logic to this function and rename it to `validateAuditLogs`?

Reply:
Sure. Do you plan to move the diff log call here as well?